packages/@react-aria/tag/src/TagKeyboardDelegate.ts

```diff
@@ -10,25 +10,25 @@
  * governing permissions and limitations under the License.
  */
 
-import {GridCollection} from '@react-types/grid';
-import {GridKeyboardDelegate} from '@react-aria/grid';
+import {Collection, Direction, KeyboardDelegate} from '@react-types/shared';
 import {Key} from 'react';
+export class TagKeyboardDelegate<T> implements KeyboardDelegate {
+  private collection: Collection<T>;
+  private direction: Direction;
 
-export class TagKeyboardDelegate<T> extends GridKeyboardDelegate<T, GridCollection<T>> {
-  getFirstKey() {
-    let key = this.collection.getFirstKey();
-    let item = this.collection.getItem(key);
+  constructor(collection: Collection<T>, direction: Direction) {
+    this.collection = collection;
+    this.direction = direction;
+  }
 
-    return [...item.childNodes][0].key;
+  getFirstKey() {
+    return this.collection.getFirstKey();
   }
 
   getLastKey() {
-    let key = this.collection.getLastKey();
-    let item = this.collection.getItem(key);
-
-    return [...item.childNodes][0].key;
+    return this.collection.getLastKey();
```

Comment:
this is likely to behave oddly when in the collapsed mode
just something to keep an eye out for, ActionGroup home/end don't do anything